src/server/api/controllers/products.controller.js

```diff
@@ -1,8 +1,16 @@
 const knex = require('../../config/db');
 const HttpError = require('../lib/utils/http-error');
 
-const getProducts = async () => {
-  return knex('products');
+const getProducts = async (request) => {
+  let products = knex('products');
+
+  if (request.query.category) {
+    products = products
+      .join('categories', 'categories.id', 'category_id')
+      .select('products.*')
+      .where('categories.name', 'like', `%${request.query.category}%`);
+  }
+  return products;
```

Comment:
To avoid nesting if statements and increasing complexity in the future (imagine we have to filter for both price and category) you can make guard clauses which will do a negative check first and then return early and otherwise do the regular logic.


```suggestion
const getProducts = async (request) => {
 if (!request.query.category) {
   return knex('products');
 }

products = products
  .join('categories', 'categories.id', 'category_id')
  .select('products.*')
  .where('categories.name', 'like', `%${request.query.category}%`);
}
return products;
```

Reply:
After discussing with @zkwsk , @simonreddy2001
To avoid merging conflict with PR #89 and this PR #86, 
the feature api/products?category={category_name} has been added to the PR #89 and will be merge only PR#89 with both features for: api/products?category={category_name} 
and api/products?daysBeforeToday={number} .